addon/hint/sql-hint.js

```diff
@@ -28,6 +29,15 @@
     return CodeMirror.resolveMode(mode).keywords;
   }
 
+  function getIdentifierQuote(editor) {
+    var mode = editor.doc.modeOption;
```

Comment:
It's easier to do `editor.getMode()` to get the mode (and then check `mode.name == "sql"`)